src/cdef.rs

```diff
@@ -41,29 +43,37 @@ wrap_fn_ptr!(pub unsafe extern "C" fn cdef(
     dst_ptr: *mut DynPixel,
     stride: ptrdiff_t,
     left: *const [LeftPixelRow2px<DynPixel>; 8],
-    top: *const DynPixel,
-    bottom: *const DynPixel,
+    top_ptr: *const DynPixel,
+    bottom_ptr: *const DynPixel,
     pri_strength: c_int,
     sec_strength: c_int,
     dir: c_int,
     damping: c_int,
     edges: CdefEdgeFlags,
     bitdepth_max: c_int,
     _dst: *const FFISafe<Rav1dPictureDataComponentOffset>,
+    _top: *const FFISafe<DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>>,
+    _bottom: *const FFISafe<CdefBottom<'_>>,
 ) -> ());
 
+#[derive(Clone, Copy)]
+pub enum CdefBottom<'a> {
+    Pic(Rav1dPictureDataComponentOffset<'a>),
+    LineBuf((&'a DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>, usize)),
```

Comment:
```suggestion
    Line {
        buf: &'a DisjointMut<AlignedVec<u8, Align64<[u8; 64]>>>,
        offset: usize,
    },
```

Reply:
Also, would it be simpler to put the `enum` only around the `DisjointMut<AlignedVec64<u8>>` and `Rav1dPictureDataComponent`, since the `offset` is the same for both variants?

Reply:
Yeah, that's reasonable and unifies the offset logic. I made that change.